cmd/juju/status/formatter.go

```diff
@@ -14,34 +14,49 @@ import (
 
 	"github.com/juju/juju/apiserver/params"
 	"github.com/juju/juju/cmd/juju/common"
+	"github.com/juju/juju/cmd/juju/storage"
 	"github.com/juju/juju/core/status"
 	"github.com/juju/juju/state/multiwatcher"
 )
 
 type statusFormatter struct {
-	status         *params.FullStatus
-	controllerName string
-	relations      map[int]params.RelationStatus
-	isoTime        bool
-	showRelations  bool
+	storage                             *storage.CombinedStorage
+	status                              *params.FullStatus
+	controllerName                      string
+	relations                           map[int]params.RelationStatus
+	isoTime, showRelations, showStorage bool
 }
 
 // NewStatusFormatter takes stored model information (params.FullStatus) and populates
 // the statusFormatter struct used in various status formatting methods
 func NewStatusFormatter(status *params.FullStatus, isoTime bool) *statusFormatter {
-	return newStatusFormatter(status, "", isoTime, true)
+	return newStatusFormatter(
+		newStatusFormatterParams{
+			status:        status,
+			isoTime:       isoTime,
+			showRelations: true,
+		})
+}
+
+type newStatusFormatterParams struct {
+	storage                             *storage.CombinedStorage
+	status                              *params.FullStatus
+	controllerName                      string
+	isoTime, showRelations, showStorage bool
 }
 
-func newStatusFormatter(status *params.FullStatus, controllerName string, isoTime, showRelations bool) *statusFormatter {
+func newStatusFormatter(p newStatusFormatterParams) *statusFormatter {
 	sf := statusFormatter{
-		status:         status,
-		controllerName: controllerName,
+		storage:        p.storage,
+		status:         p.status,
+		controllerName: p.controllerName,
 		relations:      make(map[int]params.RelationStatus),
-		isoTime:        isoTime,
-		showRelations:  showRelations,
+		isoTime:        p.isoTime,
+		showRelations:  p.showRelations,
+		showStorage:    p.showStorage,
```

Comment:
We don't need showStorage on statusFormatter - if storage is nil, that means don't show it

Reply:
I am thinking we probably still show an empty storage section if user specified `--storage` even if the storage is `nil`. And we should not show the storage section if the user didn't specify `--storage` even if the storage is not `nil`, right?

Reply:
ah, you are right. thanks!